packages/effect/src/Metric.ts

```diff
@@ -152,6 +152,10 @@ export const make: MetricApply = internal.make
  * @category mapping
  */
 export const mapInput: {
+  <Type, In, Out, In2>(
```

Comment:
I think we will want a separate api for this, otherwise you will get back a function instead of a new metric by default.

Maybe `Metric.updateMap`?

Reply:
Actually, Metric.trackSuccessWith already exists :)

Reply:
Thanks for pointing that out @tim-smart ! I think that the `A extends In2` on trackSuccessWith is still causing inference to fail. 

I'm on my phone, so I don't know if it's just a playground performance issue in my browser, but I tested here: https://effect.website/play/#3080ee0cadf4

Would you be opposed if I repurpose this PR to change the type of trackSuccessWith to remove the `A extends` and match the overload I had added originally to mapInput here which handles inference correctly?

Reply:
Fwiw I believe in this context, you want inference to pick up and fully determine In2. Is there an advantage to calling out explicitly that it is a super type of what should actually be inferred that I am missing?

Reply:
Yes I think it could be

```ts
export const trackSuccessWith: {
  <Type, In, Out, A>(
    metric: Metric<Type, In, Out>,
    f: (value: Types.NoInfer<A>) => In
  ): <A, E, R>(self: Effect.Effect<A, E, R>) => Effect.Effect<A, E, R>
  <A, E, R, Type, In, Out>(
    self: Effect.Effect<A, E, R>,
    metric: Metric<Type, In, Out>,
    f: (value: Types.NoInfer<A>) => In
  ): Effect.Effect<A, E, R>
} = internal.trackSuccessWith
```

Reply:
Curious along similar lines, any reason not to replace `unknown` with `E` in `trackDefectWith` ?

```
declare const trackDefectWith: {
  <Type, In, Out>(
    metric: Metric<Type, In, Out>,
    f: (defect: unknown) => In
  ): <A, E, R>(self: Effect.Effect<A, E, R>) => Effect.Effect<A, E, R>
  <A, E, R, Type, In, Out>(
    self: Effect.Effect<A, E, R>,
    metric: Metric<Type, In, Out>,
    f: (defect: unknown) => In
  ): Effect.Effect<A, E, R>
}
```

Reply:
Defects are not typed, so unknown makes sense there.